### PR TITLE
3D Scatter Plot Viewer in Glue

### PR DESCRIPTION
**Pull request by PennyQ**
Here is the Glue 3D scatter plot viewer with a very simple user interface(see below), which enables the property selection by the user now.
image

The questions I got now are:
- How to combine the `size` property selected by the user from the dataset with the real size showed in the viewer. In the current code we use the method from @maxwelltsai while it only works for the `mass` property.
- Any other functionalities to be implemented like color scheme or axis? 
- Here is a pretty example from the Vispy for the galaxy simulation display, maybe we could borrow some shader code from it? https://github.com/vispy/vispy/blob/master/examples/demo/gloo/galaxy/galaxy.py

Do you have any advice? Thanks and please feel free to polish the code :)

### CODE REVIEW

**Comment by PennyQ** [comment]
Here is the 3d Scatter Plot Viewer with some advanced functionality as threshold control, color picker and a small color display area. I am thinking about adding layers for current `Volume` and `Scatter` viewer to link different dataset together :) That would be the most exciting part !

image

**Comment by astrofrog** [comment]
@Penny - is this in a state where it should work? I tried it on a different catalog, and I could not see any points, but it could just be because this is in progress?

**Comment by PennyQ** [comment]
@astrofrog Yeah I had a new version of this viewer which set the position & distance of the camera according to the dataset, but another reason for the 'disappear' of those points is the size calculation we used is based on the mass of each object/point, so if users choose other properties for displaying the size then the point might be too small to show up. Max and I are still thinking about a general way to display the correct size of those points, any advice?

I will commit my code here soon today :) Thanks!

**Comment by astrofrog** [comment]
For the size, I think it makes sense to be able to select a property to use, as well as the min/max to use for normalization - so for example I could select mass, 1 to 100 msun, and points of 1msun will be the smallest size and 100msun will be the largest size, where the min/max size could be hard-coded for now (and later configurable)

**Comment by PennyQ** [comment]
The latest commit is the final code for 3D scatter plot with new features and size normalization (thanks @maxwelltsai for helping with the normalization!)

@astrofrog Please let me know if it is ok for merging, I will think about a better highlight solution for those scatter points :)

File: README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 Experimental Glue VisPy plugin
 ==============================
 
-[![Build Status](https://travis-ci.org/glue-viz/glue-3d-viewer.svg)](https://travis-ci.org/glue-viz/glue-3d-viewer?branch=master)
-[![Build status](https://ci.appveyor.com/api/projects/status/1gov2vtuesjnij69/branch/master?svg=true)](https://ci.appveyor.com/project/astrofrog/glue-3d-viewer/branch/master)
+[![Build Status](https://travis-ci.org/PennyQ/glue-3d-viewer.svg)](https://travis-ci.org/PennyQ/glue-3d-viewer?branch=master)
+[![Build status](https://ci.appveyor.com/api/projects/status/nc4ernogomn164m6?svg=true)](https://ci.appveyor.com/project/PennyQ/glue-3d-viewer)
+
```
Comment (astrofrog) [inline README.md:6]: Could you undo these changes? We don't want to merge this into master.

**Comment by PennyQ** [comment]
@astrofrog I already modified the README.md and I will merge this PR now. Is it ok?

**Comment by astrofrog** [comment]
@PennyQ - yes, we can merge this :)

**Comment by astrofrog** [comment]
I fixed the imports in 8c273de5046746561fc209910d4084d4585f1b61
